web/src/settings_realm_user_settings_defaults.js

```diff
@@ -23,22 +23,47 @@ export function maybe_disable_widgets() {
     }
 }
 
+function hide_save_discard_widget($subsection) {
+    if ($subsection.find(".save-button-controls").hasClass("hide")) {
+        return;
+    }
+
+    const properties_elements = settings_components.get_subsection_property_elements($subsection);
+    const show_change_process_button = properties_elements.some((elem) =>
+        settings_components.check_realm_default_settings_property_changed(elem),
+    );
+
+    if (show_change_process_button) {
+        // We do nothing if the selected value for any
+        // setting is not same as the original value.
+        return;
+    }
+
+    // Hide buttons if the setting was updated to the same value by other user.
+    const $save_btn_controls = $subsection.find(".save-button-controls");
+    settings_components.change_save_button_state($save_btn_controls, "discarded");
+}
+
```

Comment:
I was also thinking about duplicate code here, but then those are just one line code and I was not sure if having two separate functions for those would be worth.

Another way of doing this can be to just have a single function which would accept some argument like `setting_type` and then call the appropriate function to compute `show_change_process_button`.

Reply:
> Another way of doing this can be to just have a single function which would accept some argument like setting_type and then call the appropriate function to compute show_change_process_button.

Yeah, I think that would be better. But, I don't have too much context on the last splitting functions PR and whether we think it is the right direction or not. If we're okay with this approach, then having a single function would be fine. 

If we don't want to follow that approach, I think those 1-2 line functions might still help keep that logic in one place and we can treat those functions as `settings_components` exposing it's API.